Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -69,7 +72,111 @@ protected function getClassMetadata(ResourceInterface $resource)
      */
     protected function isPropertyEnabled($property)
     {
-        return null === $this->properties || array_key_exists($property, $this->properties);
+        $enabled = false;
+
+        if (null === $this->properties) {
+            // to ensure sanity, nested properties must still be explicitly enabled
+            $enabled = !$this->isPropertyNested($property);
```

Comment:
You can just `return` here. And then `return` the `array_key_exists` condition to have a cleaner code:
```php
if (null === $this->properties) {
    return !$this->isPropertyNested($property);
}

return array_key_exists($property, $this->properties);
```